src/CommerceModel.BetterRetail/artifacts/OOE/BetterRetail/ExecuteSQLScriptActivity/Order/AddCustomAttributes.sql

```diff
@@ -0,0 +1,530 @@
+	BEGIN
+		INSERT INTO [dbo].[ENTITY_ATTRIBUTE_LOCALIZE] (
+			Entity_Attribute_Guid
+			,CultureIso
+			,DisplayName
+		) VALUES (
+			@Entity_Attribute_Guid
+			,@CultureIso
+			,@DisplayName
+		)
+	END
+GO
+
+CREATE PROCEDURE #AddCustomAttribute
+	@Entity_Attribute_Guid	uniqueidentifier
+	,@AttributeName nvarchar(64)
+	,@DataType varchar(16)
+	,@DefaultValue nvarchar(128)
+	,@GroupName nvarchar(128)
+	,@Description nvarchar(256)
+	,@MinValue nvarchar(32)
+	,@MaxValue nvarchar(32)
+	,@IsMultilingual bit
+	,@IsSearchable bit
+	,@ReferenceLookUpName nvarchar(64)
+	,@SequenceNumber int
+	,@DataTypeSequence int	
+	,@Required bit
+	,@AllowMultipleValues bit
+AS
+
+IF NOT EXISTS(SELECT 1               
+				FROM dbo.ENTITY_ATTRIBUTE               
+				WHERE Entity_Attribute_Guid=@Entity_Attribute_Guid) 
+BEGIN
+	DECLARE @EntityName nvarchar(64) = 'Order'
+	DECLARE @FinalDataTypeSequence int = COALESCE(@DataTypeSequence,0)
+	DECLARE @FinalGroupName nvarchar(128) = COALESCE(@GroupName, 'Default')
+
+	IF (COALESCE(@DataTypeSequence,0) = 0)
+	BEGIN
+		SELECT @FinalDataTypeSequence = COALESCE(MAX(@DataTypeSequence),0) + 1 
+		FROM dbo.ENTITY_ATTRIBUTE 
+		WHERE DataType = @DataType
+		  AND EntityName LIKE @EntityName
+		  AND IsSystemAttribute = 0
+	END
+
+	DECLARE @MinMultiplicity VARCHAR(1) = '0'
+	DECLARE @MaxMultiplicity VARCHAR(1) = '1'
+
+	IF(COALESCE(@Required,0) = 1)
+	BEGIN
+		SET @MinMultiplicity = '1'
+	END
+
+	IF(COALESCE(@AllowMultipleValues,0) = 1)
+	BEGIN
+		SET @MaxMultiplicity = '*'
+	END
+
+	INSERT INTO dbo.ENTITY_ATTRIBUTE
+		(
+			EntityName
+			,AttributeName
+			,IsSystemAttribute
+			,DataType
+			,DefaultValue
+			,GroupName
+			,[Description]
+			,MinValue
+			,MaxValue
+			,MinMultiplicity
+			,MaxMultiplicity
+			,IsMultilingual
+			,IsSearchable
+			,ReferenceLookUpName
+			,SequenceNumber
+			,IsDeleted
+			,Entity_Attribute_Guid
+			,DataTypeSequence
+		)
+	VALUES
+		(
+			@EntityName
+			,@AttributeName
+			,0
+			,@DataType
+			,@DefaultValue
+			,@FinalGroupName
+			,@Description
+			,@MinValue
+			,@MaxValue
+			,@MinMultiplicity
+			,@MaxMultiplicity
+			,COALESCE(@IsMultilingual,0)
+			,COALESCE(@IsSearchable,0)
+			,@ReferenceLookUpName
+			,@SequenceNumber
+			,0
+			,@Entity_Attribute_Guid
+			,@FinalDataTypeSequence
+		);
+END
+GO
+
+/*
+
+SPECIFIC CUSTOM ATTRIBUTE STORED PROCEDURES
+
+*/
+
+CREATE PROCEDURE #AddBooleanCustomAttribute
+	@Entity_Attribute_Guid	uniqueidentifier
+	,@AttributeName nvarchar(64)
+	,@DefaultValue bit
+	,@GroupName nvarchar(128)
+	,@Description nvarchar(256)
+	,@SequenceNumber int
+	,@DataTypeSequence int
+	,@Required bit
+	,@AllowMultipleValues bit
+AS
+
+IF NOT EXISTS(SELECT 1           
+				FROM sys.columns           
+				WHERE Name = @AttributeName          
+				AND Object_ID = Object_ID(N'dbo.ORDER'))
+BEGIN
+	DECLARE @sql NVARCHAR(255) = 'ALTER TABLE dbo.[ORDER] ADD ' +  @AttributeName + ' BIT'
+	EXEC (@sql)
```

Comment:
In this case it does not change anything, for dynamic sql, it's a good habit to use sp_executesql instead of exec.